src/react/reconcilation.js

```diff
@@ -812,6 +812,53 @@ export class Reconciler {
     return value;
   }
 
+  _resolveBranchedComponentType(
+    componentType: Value,
+    reactElement: ObjectValue,
+    context: ObjectValue | AbstractObjectValue,
+    branchStatus: BranchStatusEnum,
+    branchState: BranchState | null,
+    evaluatedNode: ReactEvaluatedNode
+  ) {
+    let typeValue = getProperty(this.realm, reactElement, "type");
+    let propsValue = getProperty(this.realm, reactElement, "props");
+    let keyValue = getProperty(this.realm, reactElement, "key");
+    let refValue = getProperty(this.realm, reactElement, "ref");
+
+    const resolveBranch = (abstract: AbstractValue): AbstractValue => {
+      let condition = abstract.args[0];
+      invariant(condition instanceof AbstractValue);
+      let left = abstract.args[1];
+      let right = abstract.args[2];
+
+      if (left instanceof AbstractValue && left.kind === "conditional") {
+        left = resolveBranch(left);
```

Comment:
Is this meant to enable recursive conditionals? I tried that but it doesn't work because it fails earlier.

```js
let Type = props.switch ? (props.switch ? Foo : Bar) : Bar;
```

```
  Invariant Violation: undefined
    This is likely a bug in Prepack, not your code. Feel free to open an issue on GitHub.

      14 |   const message = `${format}
      15 | This is likely a bug in Prepack, not your code. Feel free to open an issue on GitHub.`;
    > 16 |   let error = new Error(message);
      17 |   error.name = "Invariant Violation";
      18 |   throw error;
      19 | }
      
      at invariant (src/invariant.js:16:15)
      at AbstractObjectValue.$Get (src/values/AbstractObjectValue.js:460:19)
      at Get (lib/methods/get.js:342:12)
      at Reconciler._getComponentResolutionStrategy (src/react/reconcilation.js:786:3)
      at Reconciler._resolveReactElement (src/react/reconcilation.js:1050:25)
      at Reconciler._resolveDeeply (src/react/reconcilation.js:1210:7)
      at Reconciler._renderComponent (src/react/reconcilation.js:753:5)
```

We should either fix this and add a test, or not try to handle this case.

Reply:
Will add more test cases and fix this.